diff --git a/manual_cpp b/manual_cpp
index 6f74a79..dbdfc65 100755
--- a/manual_cpp
+++ b/manual_cpp
@@ -46,7 +46,7 @@ elif len(files) == 0:
 elif len(files) == 1:
     file, = files
 
-tmp_filename = "manually_preprocessed_" + file.replace("/", "_")
+tmp_filename = "manually_preprocessed_" + file.replace("/", "_").replace("-", "_")
 
 try:
     output = args.index("-o")
diff --git a/src/elpa_impl_math_template.F90 b/src/elpa_impl_math_template.F90
index b96f219..0a9652b 100644
--- a/src/elpa_impl_math_template.F90
+++ b/src/elpa_impl_math_template.F90
@@ -1195,10 +1195,10 @@
 #endif
 #ifdef COMPLEXCASE
 #ifdef DOUBLE_PRECISION_COMPLEX
-                    & !bind(C, name="elpa_solve_tridiagonal_dc")
+                    bind(C, name="elpa_solve_tridiagonal_dc")
 #endif
 #ifdef SINGLE_PRECISION_COMPLEX
-                    & !bind(C, name="elpa_solve_tridiagonal_fc")
+                    bind(C, name="elpa_solve_tridiagonal_fc")
 #endif
 #endif
 
diff --git a/src/multiply_a_b/GPU/CUDA/mod_multiply_a_b_cuda.F90 b/src/multiply_a_b/GPU/CUDA/mod_multiply_a_b_cuda.F90
index 1cea90f..1bdf309 100644
--- a/src/multiply_a_b/GPU/CUDA/mod_multiply_a_b_cuda.F90
+++ b/src/multiply_a_b/GPU/CUDA/mod_multiply_a_b_cuda.F90
@@ -61,18 +61,6 @@ module multiply_a_b_cuda
 
   public
 
-  interface
-    subroutine cuda_copy_double_tmp2_c_intptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
-                                                                     ldc, ldcCols, my_stream)&
-                                                     bind(C, name="cuda_copy_double_tmp2_c_FromC")
-      use, intrinsic :: iso_c_binding
-      implicit none
-      integer(kind=C_intptr_T), value  :: tmp2_dev, c_dev
-      integer(kind=c_int), intent(in)  :: nr_done, nstor, lcs, lce, ldc, ldcCols
-      integer(kind=c_intptr_t), value  :: my_stream
-    end subroutine
-  end interface
-
   interface
     subroutine cuda_copy_double_tmp2_c_cptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
                                                                      ldc, ldcCols, my_stream)&
diff --git a/src/multiply_a_b/GPU/OpenMP/mod_multiply_a_b_openmp_offload.F90 b/src/multiply_a_b/GPU/OpenMP/mod_multiply_a_b_openmp_offload.F90
index 5c2bb00..0d4f50f 100644
--- a/src/multiply_a_b/GPU/OpenMP/mod_multiply_a_b_openmp_offload.F90
+++ b/src/multiply_a_b/GPU/OpenMP/mod_multiply_a_b_openmp_offload.F90
@@ -61,18 +61,6 @@ module multiply_a_b_openmp_offload
 
   public
 
-  interface
-    subroutine openmp_offload_copy_double_tmp2_c_intptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
-                                                                     ldc, ldcCols, my_stream)&
-                                                     bind(C, name="openmp_offload_copy_double_tmp2_c_FromC")
-      use, intrinsic :: iso_c_binding
-      implicit none
-      integer(kind=C_intptr_T), value  :: tmp2_dev, c_dev
-      integer(kind=c_int), intent(in)  :: nr_done, nstor, lcs, lce, ldc, ldcCols
-      integer(kind=c_intptr_t), value  :: my_stream
-    end subroutine
-  end interface
-
   interface
     subroutine openmp_offload_copy_double_tmp2_c_cptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
                                                                      ldc, ldcCols, my_stream)&
diff --git a/src/multiply_a_b/GPU/ROCm/mod_multiply_a_b_hip.F90 b/src/multiply_a_b/GPU/ROCm/mod_multiply_a_b_hip.F90
index 6198618..e7bc01b 100644
--- a/src/multiply_a_b/GPU/ROCm/mod_multiply_a_b_hip.F90
+++ b/src/multiply_a_b/GPU/ROCm/mod_multiply_a_b_hip.F90
@@ -61,18 +61,6 @@ module multiply_a_b_hip
 
   public
 
-  interface
-    subroutine hip_copy_double_tmp2_c_intptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
-                                                                     ldc, ldcCols, my_stream)&
-                                                     bind(C, name="hip_copy_double_tmp2_c_FromC")
-      use, intrinsic :: iso_c_binding
-      implicit none
-      integer(kind=C_intptr_T), value  :: tmp2_dev, c_dev
-      integer(kind=c_int), intent(in)  :: nr_done, nstor, lcs, lce, ldc, ldcCols
-      integer(kind=c_intptr_t), value  :: my_stream
-    end subroutine
-  end interface
-
   interface
     subroutine hip_copy_double_tmp2_c_cptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
                                                                      ldc, ldcCols, my_stream)&
diff --git a/src/multiply_a_b/GPU/SYCL/mod_multiply_a_b_sycl.F90 b/src/multiply_a_b/GPU/SYCL/mod_multiply_a_b_sycl.F90
index 1b87e97..1d3f22e 100644
--- a/src/multiply_a_b/GPU/SYCL/mod_multiply_a_b_sycl.F90
+++ b/src/multiply_a_b/GPU/SYCL/mod_multiply_a_b_sycl.F90
@@ -61,18 +61,6 @@ module multiply_a_b_sycl
 
   public
 
-  interface
-    subroutine sycl_copy_double_tmp2_c_intptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
-                                                                     ldc, ldcCols, my_stream)&
-                                                     bind(C, name="sycl_copy_double_tmp2_c_FromC")
-      use, intrinsic :: iso_c_binding
-      implicit none
-      integer(kind=C_intptr_T), value  :: tmp2_dev, c_dev
-      integer(kind=c_int), intent(in)  :: nr_done, nstor, lcs, lce, ldc, ldcCols
-      integer(kind=c_intptr_t), value  :: my_stream
-    end subroutine
-  end interface
-
   interface
     subroutine sycl_copy_double_tmp2_c_cptr_c(tmp2_dev, c_dev, nr_done, nstor, lcs, lce, &
                                                                      ldc, ldcCols, my_stream)&
diff --git a/test/Fortran/test_autotune.F90 b/test/Fortran/test_autotune.F90
index eb14d70..6328713 100644
--- a/test/Fortran/test_autotune.F90
+++ b/test/Fortran/test_autotune.F90
@@ -312,8 +312,8 @@ program test
      status = check_correctness_analytic(na, nev, ev, z, nblk, myid, np_rows, np_cols, my_prow, my_pcol, &
                                          .true., .true., print_times=.false.)
      a(:,:) = as(:,:)
-     call e%autotune_print_state(tune_state)
-     call e%autotune_save_state(tune_state, "saved_state_"//trim(iter_string)//".txt")
+     call e%autotune_print_state(tune_state, error_elpa)
+     call e%autotune_save_state(tune_state, "saved_state_"//trim(iter_string)//".txt", error_elpa)
 
 #ifdef WITH_MPI
      ! barrier after store settings, file created from one MPI rank only, but loaded everywhere
